NitroxServer/Communication/Packets/Processors/WorldSending/AskBatchCellProcessor.cs

```diff
@@ -0,0 +1,68 @@
+﻿using NitroxModel.DataStructures;
+using NitroxModel.Packets;
+using NitroxModel.Packets.WorldSending;
+using NitroxServer.Communication.Packets.Processors.Abstract;
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace NitroxServer.Communication.Packets.Processors.WorldSending
+{
+    public class AskBatchCellProcessor : AuthenticatedPacketProcessor<AskBatchCell>
+    {
+        private TcpServer tcpServer;
+        private Dictionary<string, Queue<Chunk>> batchQueues = new Dictionary<string, Queue<Chunk>>();
+
+        public static readonly string folderName = "DefaultSave"; //todo read from config
+
+        public AskBatchCellProcessor(TcpServer tcpServer)
+        {
+            this.tcpServer = tcpServer;
+        }
+
+        public override void Process(AskBatchCell packet, Player player)
+        {
+            if (packet.FromChunkQueue)
+            {
+                ProcessFromQueue(player);
+            }
+            else
+            {
+                ProcessFromChunk(packet, player);
+            }
+        }
+
+        public void ProcessFromQueue(Player player)
+        {
+            lock (player.chunkQueues)
+            {
+                Queue<Chunk> batchQueue = player.chunkQueues;
+                if (batchQueue.Count == 0)
+                {
+                    foreach (string file in Directory.GetFiles(folderName + "\\"))
+                    {
+                        string[] chunkName = Path.GetFileNameWithoutExtension(file).Split('-');
+                        Int3 chunkCoords = new Int3(int.Parse(chunkName[2]), int.Parse(chunkName[3]), int.Parse(chunkName[4]));
+                        byte[] data = File.ReadAllBytes(file);
+                        batchQueue.Enqueue(new Chunk(chunkCoords, data));
+                    }
+                }
+
+                ReceiveBatchCell chunkPacket = new ReceiveBatchCell(player.Id, batchQueue.Peek().chunkData, batchQueue.Peek().chunkCoord, batchQueue.Count > 1);
+                batchQueue.Dequeue();
```

Comment:
I'm debating on if the queue is sparing the player and headache.  In my mind this is what will end up happening: 

1) Player will request batches 
2) Server will send one 
3) Player will Process
4) Server will send one
5) Player will process

If the players were on LAN then the latency between the player requesting a packet and the server sending it will be close to zero.  That means that the player would be receiving a chunk to process in the next frame.  Effectively equivalent to just sending all of the chunks at once.  

I guess the benefit would be that you at least get a frame in and on higher latencies you can get many frames in.  Probably worth having after thinking out loud.

Reply:
There were problems sending chunks to the client at one time. There could be a fix for that but I guess I just went with this as I was too lazy to do otherwise